compiler/rustc_interface/src/passes.rs

```diff
@@ -666,11 +647,12 @@ fn write_out_deps(
                     .emit_artifact_notification(&deps_filename, "dep-info");
             }
         }
-        Err(e) => sess.fatal(&format!(
-            "error writing dependencies to `{}`: {}",
-            deps_filename.display(),
-            e
-        )),
+        Err(e) => {
+            sess.emit_fatal(ErrorWritingDependencies {
+                path: (&deps_filename.display()).into(),
```

Comment:
See #100754 for how to avoid needing to use `DiagnosticArgFromDisplay` for paths and input errors.

Reply:
I see, I'll wait for #100754 being merged before pushing the changes.

Reply:
Feel free to just copy the commit and whoever needs to can rebase.